packages/relay/test/checkReputation.test.ts

```diff
@@ -1,69 +1,77 @@
-import { UserState } from '@unirep/core'
+import { UnirepApp } from '@unirep-app/contracts/typechain-types'
 import { stringifyBigInts } from '@unirep/utils'
+import { DB } from 'anondb'
 import { expect } from 'chai'
 import { ethers } from 'hardhat'
 import { jsonToBase64 } from '../src/middlewares/CheckReputationMiddleware'
-import { userService } from '../src/services/UserService'
 import { UnirepSocialSynchronizer } from '../src/services/singletons/UnirepSocialSynchronizer'
+import { userService } from '../src/services/UserService'
 import { Post } from '../src/types'
 import { deployContracts, startServer, stopServer } from './environment'
-import { UserStateFactory } from './utils/UserStateFactory'
-import { ReputationType, genProveReputationProof } from './utils/genProof'
+import { genProveReputationProof, ReputationType } from './utils/genProof'
 import { post } from './utils/post'
-import { signUp } from './utils/signUp'
+import { IdentityObject } from './utils/types'
+import { createRandomUserIdentity, genUserState } from './utils/userHelper'
 
 describe('CheckReputation', function () {
     let snapshot: any
     let express: ChaiHttp.Agent
-    let userState: UserState
-    let sync: UnirepSocialSynchronizer
     let chainId: number
+    let user: IdentityObject
+    let prover: any
+    let app: UnirepApp
+    let db: DB
+    let sync: UnirepSocialSynchronizer
 
     const EPOCH_LENGTH = 100000
 
     before(async function () {
         snapshot = await ethers.provider.send('evm_snapshot', [])
         // deploy contracts
-        const { unirep, app } = await deployContracts(EPOCH_LENGTH)
+        const { unirep, app: _app } = await deployContracts(EPOCH_LENGTH)
         // start server
-        const { db, prover, provider, synchronizer, chaiServer } =
-            await startServer(unirep, app)
+        const {
+            db: _db,
+            prover: _prover,
+            provider,
+            synchronizer,
+            chaiServer,
+        } = await startServer(unirep, _app)
         express = chaiServer
+        prover = _prover
+        app = _app
+        db = _db
         sync = synchronizer
 
-        const userStateFactory = new UserStateFactory(
-            db,
-            provider,
-            prover,
-            unirep,
-            app,
-            synchronizer
-        )
-
-        // initUserStatus
-        let initUser = await userService.getLoginUser(db, '123', undefined)
-        const wallet = ethers.Wallet.createRandom()
-        userState = await signUp(
-            initUser,
-            userStateFactory,
-            userService,
-            synchronizer,
-            wallet
+        user = createRandomUserIdentity()
+        const userState = await genUserState(user.id, app, db, prover)
+        const { publicSignals, _snarkProof: proof } =
+            await userState.genUserSignUpProof()
```

Comment:
I think we can discuss this more. In my understanding, we will create a custom function to generate the zk proof is because we have to mock the fake proof. But if we just test the correct case, it's fine to use common pattern?

Reply:
I see what you mean. If the signup test has passed I think in other test case we can assume it work and just use the src functions.